### PR TITLE
:sparkles: Added the Jira Workflow APIS

### PR DESCRIPTION
**Pull request by ctreminiom**

### CODE REVIEW

File: jira/internal/workflow_impl.go [hunk 1]
```diff
@@ -74,11 +74,186 @@ func (w *WorkflowService) Delete(ctx context.Context, workflowID string) (*model
...
+}
+
+func (w *WorkflowService) Updates(ctx context.Context, payload *model.WorkflowUpdatesPayloadScheme, expand []string) (*model.WorkflowUpdateResponseScheme, *model.ResponseScheme, error) {
+	return w.internalClient.Updates(ctx, payload, expand)
+}
+
+func (w *WorkflowService) ValidateUpdateWorkflows(ctx context.Context, payload *model.ValidationOptionsForUpdateScheme) (*model.WorkflowValidationErrorListScheme, *model.ResponseScheme, error) {
+	return w.internalClient.ValidateUpdateWorkflows(ctx, payload)
+}
+
```
Comment (Fank) [inline jira/internal/workflow_impl.go:100]: Please add documentation

File: pkg/infra/models/jira_workflow.go [hunk 2]
```diff
@@ -36,15 +36,50 @@ type WorkflowPublishedIDScheme struct {
...
-	Name        string                          `json:"name,omitempty"`        // The name of the transition.
-	Description string                          `json:"description,omitempty"` // The description of the transition.
-	From        []string                        `json:"from,omitempty"`        // The statuses from which this transition can be executed.
-	To          string                          `json:"to,omitempty"`          // The status to which this transition goes.
-	Type        string                          `json:"type,omitempty"`        // The type of the transition.
-	Screen      *WorkflowTransitionScreenScheme `json:"screen,omitempty"`      // The screen associated with the transition.
-	Rules       *WorkflowTransitionRulesScheme  `json:"rules,omitempty"`       // The rules of the transition.
+	Actions            []*WorkflowRuleConfigurationScheme `json:"actions,omitempty"`
+	Conditions         *ConditionGroupConfigurationScheme `json:"conditions,omitempty"`
+	CustomIssueEventId string                             `json:"customIssueEventId,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:44]:
```suggestion
	CustomIssueEventID string                             `json:"customIssueEventId,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 3]
```diff
@@ -36,15 +36,50 @@ type WorkflowPublishedIDScheme struct {
...
+	To                 *WorkflowStatusAndPortScheme       `json:"to,omitempty"`   // The status to which this transition goes.
+	ToStatusReference  string                             `json:"toStatusReference,omitempty"`
+	TransitionScreen   *WorkflowRuleConfigurationScheme   `json:"transitionScreen,omitempty"`
+	Triggers           []*WorkflowTriggerScheme           `json:"triggers,omitempty"`
+	Type               string                             `json:"type,omitempty"` // The type of the transition.
+	Validators         []*WorkflowRuleConfigurationScheme `json:"validators,omitempty"`
+}
+
+type WorkflowRuleConfigurationScheme struct {
+	Id      string `json:"id,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:59]:
```suggestion
	ID      string `json:"id,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 4]
```diff
@@ -36,15 +36,50 @@ type WorkflowPublishedIDScheme struct {
...
+}
+
+type WorkflowTransitionLinkScheme struct {
+	FromPort            int    `json:"fromPort,omitempty"`
+	FromStatusReference string `json:"fromStatusReference,omitempty"`
+	ToPort              int    `json:"toPort,omitempty"`
+}
+
+type WorkflowTriggerScheme struct {
+	Id      string `json:"id,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:81]:
```suggestion
	ID      string `json:"id,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 5]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
 	Operator      string                     `json:"operator,omitempty"`      // The operator of the condition.
 	Type          string                     `json:"type,omitempty"`          // The type of the condition.
 }
+
+type WorkflowSearchCriteria struct {
+	ProjectAndIssueTypes []*WorkflowSearchProjectIssueTypeMapping `json:"projectAndIssueTypes,omitempty"`
+	WorkflowIds          []string                                 `json:"workflowIds,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:164]:
```suggestion
	WorkflowIDs          []string                                 `json:"workflowIds,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 6]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
...
+	WorkflowNames        []string                                 `json:"workflowNames,omitempty"`
+}
+
+type WorkflowSearchProjectIssueTypeMapping struct {
+	IssueTypeId string `json:"issueTypeId,omitempty"`
+	ProjectId   string `json:"projectId,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:170]:
```suggestion
	IssueTypeID string `json:"issueTypeId,omitempty"`
	ProjectID   string `json:"projectId,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 7]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
...
+}
+
+type WorkflowReadResponseScheme struct {
+	Statuses  []*WorkflowStatusDetailScheme `json:"statuses,omitempty"`
+	Workflows []*JiraWorkflowScheme         `json:"workflows,omitempty"`
+}
+
+type JiraWorkflowScheme struct {
+	Description      string                           `json:"description,omitempty"`
+	Id               string                           `json:"id,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:180]:
```suggestion
	ID               string                           `json:"id,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 8]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
...
+	IsEditable       bool                             `json:"isEditable,omitempty"`
+	Name             string                           `json:"name,omitempty"`
+	Scope            *WorkflowStatusScopeScheme       `json:"scope,omitempty"`
+	StartPointLayout *WorkflowLayoutScheme            `json:"startPointLayout,omitempty"`
+	Statuses         []*WorkflowReferenceStatusScheme `json:"statuses,omitempty"`
+	TaskId           string                           `json:"taskId,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:186]:
```suggestion
	TaskID           string                           `json:"taskId,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 9]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
...
+	EditorScope  string                                `json:"editorScope,omitempty"`
+	ForgeRules   []*AvailableWorkflowForgeRuleScheme   `json:"forgeRules,omitempty"`
+	ProjectTypes []string                              `json:"projectTypes,omitempty"`
+	SystemRules  []*AvailableWorkflowSystemRuleScheme  `json:"systemRules,omitempty"`
+	TriggerRules []*AvailableWorkflowTriggers          `json:"triggerRules,omitempty"`
+}
+
+type AvailableWorkflowConnectRuleScheme struct {
+	AddonKey    string `json:"addonKey,omitempty"`
+	CreateUrl   string `json:"createUrl,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:219]:
```suggestion
	CreateURL   string `json:"createUrl,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 10]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
...
+	Description string `json:"description,omitempty"`
+	EditUrl     string `json:"editUrl,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:221]:
```suggestion
	EditURL     string `json:"editUrl,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 11]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
...
+	ModuleKey   string `json:"moduleKey,omitempty"`
+	Name        string `json:"name,omitempty"`
+	RuleKey     string `json:"ruleKey,omitempty"`
+	RuleType    string `json:"ruleType,omitempty"`
+	ViewUrl     string `json:"viewUrl,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:226]:
```suggestion
	ViewURL     string `json:"viewUrl,omitempty"`
```

File: pkg/infra/models/jira_workflow.go [hunk 12]
```diff
@@ -123,3 +158,262 @@ type WorkflowConditionScheme struct {
...
+
+type StatusLayoutUpdateScheme struct {
+	Layout          *WorkflowLayoutScheme `json:"layout,omitempty"`
+	StatusReference string                `json:"statusReference"`
+}
+
+type TransitionUpdateDTOScheme struct {
+	Actions            []*WorkflowRuleConfigurationScheme `json:"actions,omitempty"`
+	Conditions         *ConditionGroupUpdateScheme        `json:"conditions,omitempty"`
+	CustomIssueEventId string                             `json:"customIssueEventId,omitempty"`
```
Comment (Fank) [inline pkg/infra/models/jira_workflow.go:309]:
```suggestion
	CustomIssueEventID string                             `json:"customIssueEventId,omitempty"`
```

File: service/jira/workflow.go [hunk 13]
```diff
@@ -42,6 +42,18 @@ type WorkflowConnector interface {
...
+
+	Capabilities(ctx context.Context, workflowID, projectID, issueTypeID string) (*model.WorkflowCapabilitiesScheme, *model.ResponseScheme, error)
+
+	Creates(ctx context.Context, payload *model.WorkflowCreatesPayload) (*model.WorkflowCreateResponseScheme, *model.ResponseScheme, error)
+
+	ValidateCreateWorkflows(ctx context.Context, payload *model.ValidationOptionsForCreateScheme) (*model.WorkflowValidationErrorListScheme, *model.ResponseScheme, error)
+
+	Updates(ctx context.Context, payload *model.WorkflowUpdatesPayloadScheme, expand []string) (*model.WorkflowUpdateResponseScheme, *model.ResponseScheme, error)
+
+	ValidateUpdateWorkflows(ctx context.Context, payload *model.ValidationOptionsForUpdateScheme) (*model.WorkflowValidationErrorListScheme, *model.ResponseScheme, error)
```
Comment (Fank) [inline service/jira/workflow.go:56]: Please add documentation

**Review by Fank — CHANGES_REQUESTED** [review]

File: jira/internal/workflow_impl.go (hunk 1 above)
Comment (ctreminiom) [reply jira/internal/workflow_impl.go]: Done @Fank!

File: service/jira/workflow.go (hunk 13 above)
Comment (ctreminiom) [reply service/jira/workflow.go]: Done @Fank!

**Pull request merged**